client/apis/objectstorage/v1alpha2/bucket_types.go

```diff
@@ -58,52 +59,68 @@ type BucketSpec struct {
 	// parameters is an opaque map of driver-specific configuration items passed to the driver that
 	// fulfills requests for this Bucket.
 	// +optional
+	// +kubebuilder:validation:MinProperties=1
 	// +kubebuilder:validation:XValidation:message="parameters map is immutable",rule="self == oldSelf"
 	Parameters map[string]string `json:"parameters,omitempty"`
 
 	// protocols lists object store protocols that the provisioned Bucket must support.
 	// If specified, COSI will verify that each item is advertised as supported by the driver.
+	// Possible values: 'S3', 'Azure', 'GCS'.
 	// +optional
 	// +listType=set
 	// +kubebuilder:validation:XValidation:message="protocols list is immutable",rule="self == oldSelf"
 	Protocols []ObjectProtocol `json:"protocols,omitempty"`
```

Comment:
@JoelSpeed you recommended `MinProperties` for maps. Is there an equivalent recommendation for `MinItems` for lists, or are maps more susceptible to concerns?

Reply:
I'd recommend a minimum on both maps and lists yes, in fact, any type

What you want to avoid is someone persisting the empty object when semantically there is no difference between unset and empty, so in this case, adding a MinItems=1 would be advised

KAL has `minlength` which should point out where these are missing for you